### PR TITLE
Add visa letter template and generation script

### LINKED ISSUE #74: Draft visa invitation letter
https://www.canada.ca/en/immigration-refugees-citizenship/services/visit-canada/business/bring-business-guests/organize-international-event-canada.html

A letter that event organizers need to give delegates when they apply for a visa to come to Canada.

The letter should include information on the event to establish:
- the delegate’s purpose of travel to Canada, and
- plans once in Canada.
- The letter should also indicate if financial assistance is given to delegates, such as airline tickets and accommodations.

### PR DESCRIPTION
**Pull request by benhylau**
Resolves #74

### CODE REVIEW

File: visas/info.yaml.template [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+sponsor:
+  name: ""
+  description: ""
+embassy:
+  address: ""
+  country: ""
+  phone: ""
+organizer:
+  name: ""
+  prefix: ""
```
Comment (benhylau) [inline visas/info.yaml.template]:
```suggestion
```

File: visas/letter.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+If the travel visa is approved, ##applicant.prefix## ##applicant.name## plans to arrive to Toronto a few days before the conference days and stay a couple days afterwards, depending on flight ticket arrangements. The total duration will fall within a two week period within September of 2019, given the conference is held from September 20-22. ##applicant.prefix## ##applicant.name## is a genuine visitor who is visiting Canada to participate at Our Networks 2019 and will not overstay ##applicant.pronoun## visa. Accomodation for the duration of ##applicant.pronoun## stay is arranged at my primary residence: ##organizer.address##
+
+Feel free to contact me should you have any further questions regarding myself or ##applicant.prefix## ##applicant.name##.
```
Comment (benhylau) [inline visas/letter.md]:
```suggestion
Feel free to contact me should you have any further questions regarding myself or ##applicant.prefix## ##applicant.name##. You can also contact ##applicant.prefix## ##applicant.name## directly at ##applicant.phone## and ##applicant.email##
```

File: visas/letter.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+This letter confirms that ##applicant.prefix## ##applicant.name##, who is ##applicant.nationality##, born on ##applicant.dob## and currently living in ##applicant.address##, ##applicant.country##, is invited to participate at Our Networks 2019, held from September 20–22, 2019 at the Toronto Media Arts Centre. Details about the event can be found at our conference website: https://ournetworks.ca
+
+##applicant.prefix## ##applicant.name## is currently employed as ##affiliation.title## by ##affiliation.name## in ##affiliation.country##. Our Networks is inviting ##applicant.prefix## ##applicant.name## to our conference to share ##applicant.pronoun## expertise and participate without monetary compensations from Our Networks. The cost of this trip, including flights, accomodation, and subsistence costs for the duration of ##applicant.pronoun## stay in Toronto, is covered by ##sponsor.name##, ##sponsor.description##.
```
Comment (benhylau) [inline visas/letter.md]:
```suggestion
##applicant.prefix## ##applicant.name## is currently employed as ##affiliation.title## by ##affiliation.name## in ##affiliation.country##. Our Networks is inviting ##applicant.prefix## ##applicant.name## to our conference to share ##applicant.pronoun## expertise and participate without monetary compensations from Our Networks. The cost of this trip, including flights, accommodation, and subsistence costs for the duration of ##applicant.pronoun## stay in Toronto, is covered by ##sponsor.name##, ##sponsor.description##.
```

File: visas/letter.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+If the travel visa is approved, ##applicant.prefix## ##applicant.name## plans to arrive to Toronto a few days before the conference days and stay a couple days afterwards, depending on flight ticket arrangements. The total duration will fall within a two week period within September of 2019, given the conference is held from September 20-22. ##applicant.prefix## ##applicant.name## is a genuine visitor who is visiting Canada to participate at Our Networks 2019 and will not overstay ##applicant.pronoun## visa. Accomodation for the duration of ##applicant.pronoun## stay is arranged at my primary residence: ##organizer.address##
```
Comment (benhylau) [inline visas/letter.md]:
```suggestion
If the travel visa is approved, ##applicant.prefix## ##applicant.name## plans to arrive to Toronto a few days before the conference days and stay a couple days afterwards, depending on flight ticket arrangements. The total duration will fall within a two week period within September of 2019, given the conference is held from September 20-22. ##applicant.prefix## ##applicant.name## is a genuine visitor who is visiting Canada to participate at Our Networks 2019 and will not overstay ##applicant.pronoun## visa. Accommodation for the duration of ##applicant.pronoun## stay is arranged at my primary residence: ##organizer.address##
```

File: visas/letter.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+To the Canadian Embassy in ##embassy.country##
+
+##embassy.address##
+##embassy.phone##
+
+**Subject: Invitation Letter for ##applicant.prefix## ##applicant.name##**
+
+Dear Sir/Madam,
+
+My name is ##organizer.name##, I was born on ##organizer.dob##, and I am from Toronto, Canada. I am a Canadian citizen and I live in ##organizer.address##, ##organizer.country##. I am a co-organizer of an annual conference in Toronto called Our Networks, which explores our technology practices and collective maintenance of our network infrastructures.
```
Comment (dcwalk) [inline visas/letter.md]:
```suggestion
My name is ##organizer.name##, I was born on ##organizer.dob##, and I am from Toronto, Canada. I am a Canadian citizen and I live in ##organizer.address##, ##organizer.country##. I am a co-organizer of an annual conference in Toronto called Our Networks, which explores networking technology practices and digital society.
```

File: visas/letter.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+This letter confirms that ##applicant.prefix## ##applicant.name##, who is ##applicant.nationality##, born on ##applicant.dob## and currently living in ##applicant.address##, ##applicant.country##, is invited to participate at Our Networks 2019, held from September 20–22, 2019 at the Toronto Media Arts Centre. Details about the event can be found at our conference website: https://ournetworks.ca
```
Comment (dcwalk) [inline visas/letter.md]:
```suggestion
This letter confirms that ##applicant.prefix## ##applicant.name##, who is ##applicant.nationality##, born on ##applicant.dob## and currently living in ##applicant.address##, ##applicant.country##, is invited to participate at Our Networks 2019. The conference will be held from September 20–22, 2019 at the Toronto Media Arts Centre, details about the event can be found on our website: https://ournetworks.ca.
```

File: visas/letter.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+##applicant.prefix## ##applicant.name## is currently employed as ##affiliation.title## by ##affiliation.name## in ##affiliation.country##. Our Networks is inviting ##applicant.prefix## ##applicant.name## to our conference to share ##applicant.pronoun## expertise and participate without monetary compensations from Our Networks. The cost of this trip, including flights, accommodation, and subsistence costs for the duration of ##applicant.pronoun## stay in Toronto, is covered by ##sponsor.name##, ##sponsor.description##.
```
Comment (dcwalk) [inline visas/letter.md]:
```suggestion
##applicant.prefix## ##applicant.name## is currently employed as ##affiliation.title## by ##affiliation.name## in ##affiliation.country##. Our Networks is inviting ##applicant.prefix## ##applicant.name## to our conference to share ##applicant.pronoun## expertise and participate without monetary compensation from Our Networks. The cost of the trip, including flights, accommodation, and subsistence costs for the duration of ##applicant.pronoun## stay in Toronto, is covered by ##sponsor.name##, ##sponsor.description##.
```

File: visas/letter.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+If the travel visa is approved, ##applicant.prefix## ##applicant.name## plans to arrive to Toronto a few days before the conference days and stay a couple days afterwards, depending on flight ticket arrangements. The total duration will fall within a two week period within September of 2019, given the conference is held from September 20-22. ##applicant.prefix## ##applicant.name## is a genuine visitor who is visiting Canada to participate at Our Networks 2019 and will not overstay ##applicant.pronoun## visa. Accommodation for the duration of ##applicant.pronoun## stay is arranged at my primary residence: ##organizer.address##
```
Comment (dcwalk) [inline visas/letter.md]:
```suggestion
If the travel visa is approved, ##applicant.prefix## ##applicant.name## plans to arrive to Toronto a few days before the conference and stay a couple days afterward, depending on flight arrangements. The total duration will fall within a two week period in September 2019, given that the conference is September 20-22. ##applicant.prefix## ##applicant.name## is a genuine visitor who is visiting Canada to participate at Our Networks 2019 and will not overstay ##applicant.pronoun## visa. Accommodation for the duration of ##applicant.pronoun## stay is arranged at my primary residence: ##organizer.address##.
```

File: visas/letter.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+Feel free to contact me should you have any further questions regarding myself or ##applicant.prefix## ##applicant.name##. You can also contact ##applicant.prefix## ##applicant.name## directly at ##applicant.phone## and ##applicant.email##
```
Comment (dcwalk) [inline visas/letter.md]:
```suggestion
Feel free to contact me should you have any further questions regarding myself or ##applicant.prefix## ##applicant.name##. You can also contact ##applicant.prefix## ##applicant.name## directly at ##applicant.phone## and ##applicant.email##.
```

File: visas/info.yaml.sample [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  description: "her employer"
+embassy:
+  address: "200 Other St., Other City"
+  country: "Scotland"
+  phone: "234-567-890"
+organizer:
+  name: "Benedict Lau"
+  prefix: "Mr."
+  dob: "July 1, 1980"
+  address: "400 Some Ave., Toronto"
```
Comment (dcwalk) [inline visas/info.yaml.sample]: Do you need Province + Post Code?

File: visas/letter.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+Kind regards,
+
+![signature](./signature.png)
+##organizer.name##
+Our Networks
```
Comment (dcwalk) [inline visas/letter.md]: I think you can cut this? Also not the official organizer address below, and you are sending letter as individual

Comment (dcwalk) [inline visas/letter.md]:
```suggestion

```

File: visas/info.yaml.sample (hunk 10 above)
Comment (benhylau) [reply visas/info.yaml.sample]: Yes, will add to sample and use full address.

Comment (benhylau) [reply visas/info.yaml.sample]: Yes, will add to sample and use full address.

Comment (benhylau) [inline visas/info.yaml.sample]:
```suggestion
  address: "400 Some Ave., Toronto, ON A1B 2C3"
```

**Comment by benhylau** [comment]
Letters sent out, pls merge if this looks okay

**Comment by dcwalk** [comment]
agree on `.yaml` -> `.yml`

**Comment by dcwalk** [comment]
Just pasting notes here about letter that was sent by email to applicants, I think we might want an email template for future years? (Or upcoming applicants?)

--

To write it, we need your:

Date of birth
Contact information (telephone number and email)
Exact address
The purpose of the visit to Canada (if you're doing anything besides attend Our Networks)
The duration of the visit
Where you will stay in Canada and how you will pay for the stay (probably the name of your employer and job title)
The date when you will arrive in Canada and the date you will leave (approximately is probably fine, since we know the conference dates)

**Comment by dcwalk** [comment]
Okay, took care of my own itches above as well as tested.

Does it look okay @benhylau?

**Comment by benhylau** [comment]
@dcwalk I added some fields to intial email based on what we actually need after all the exchanges. Please go ahead and merge if looks okay.

**Comment by dcwalk** [comment]
looks good, minor tweak on the initial email but only cosmetic.

**Pull request merged**